core/jobs/batch_jobs/math_interactions_audit_jobs.py

```diff
@@ -104,9 +104,9 @@ def contains_math_interactions(
         Returns:
             bool. Whether the exploration contains math interactions.
         """
-        # We are adding no branch here because the any() method returns either
-        # True or False but the case where it exits out of execution will never
-        # occur. Hence we add the flag to ignore the branch (110 -> exit).
+        # The method any() returns either True or False. The case where the
```

Comment:
Given this explanation, I still don't understand why we need "no branch" here. What happens when it's removed? I don't see anything that implies that "method doesn't return anything" is a case that should even be considered.

Reply:
Coverage tool does not realize that `any()` method surely returns (either True or False). See [docs](https://docs.python.org/3/library/functions.html#any). It thinks of a branch when `any()` method won't return anything which will never arise. Hence, we have to use the `no branch` flag to deal with the hypothetical branch when the method simply exits without returning anything.

Reply:
How did you draw the conclusion "It thinks of a branch when any() method won't return anything which will never arise"? Can you link to references or proof of this?

`any` is such a basic construct that it seems odd that coverage behaves in this way, so in the absence of evidence, my prior feeling is that there is something else going on here.

Reply:
I investigated into how `any()` method is implemented behind the scenes and replaced it with that part of code (Using a `for loop` and an `If condition`). Seems like there a was a branch exiting out of the for loop which wasn't been accounted for.

Removed this comment and added new tests. PTAL. Thanks!